Methods experimental design (#11)
greenelab/text_mined_hetnet_manuscript: content/03.methods.md

### Discriminator Model Calibration
talk about calibrating deep learning models with temperature smoothing

### Experimental Design
Being able to re-use label functions across edge types would substantially reduce the number of label functions required to extract multiple relationship types from biomedical literature.
We first established a baseline by training a generative model using only distant supervision label functions designed for the target edge type.
As an example, for the gene-interacts-gene edge type we used label functions that returned a `1` if the pair of genes were included in the Human Interaction database [@doi:10.1016/j.cell.2014.10.050], the iRefIndex database [@doi:10.1186/1471-2105-9-405] or in the Incomplete Interactome database [@doi:10.1126/science.1257601].
Then we compared models that also included text and domain-heuristic label functions.
Using a sampling with replacement approach, we sampled these text and domain-heuristic label functions separately within edge types, across edge types, and from a pool of all label functions.
We compared within-edge-type performance to across-edge-type and all-edge-type performance.

> For each edge-type we sampled a fixed number of label functions consisting of five evenly-spaced numbers between one and the total number of possible label functions.

Reviewer: ```suggestion
For each edge type we sampled a fixed number of label functions consisting of five evenly-spaced numbers between one and the total number of possible label functions.
```

> This sampling process was repeated 50 times for each point.

Reviewer: ```suggestion
We repeated this sampling process 50 times for each point.
```

> At each point we evaluated both generative and discriminative models, and we report performance of each in terms of the area under the receiver operating characteristic curve (AUROC) and the area under the precision-recall curve (AUPR).

Reviewer: ```suggestion
We evaluated both generative and discriminative models at each point, and we report performance of each in terms of the area under the receiver operating characteristic curve (AUROC) and the area under the precision-recall curve (AUPR).
```

talk about calibrating deep learning models with temperature smoothing

## Experimental Design

> Being able to re-use label functions would significantly speed up the annotation process.

Reviewer 2: "re-use label functions" in what way? I believe you mean reuse label functions for different edge types?

> Being able to re-use label functions would significantly speed up the annotation process.

Reviewer: ```suggestion
Being able to re-use label functions across edge types would substantially reduce the number of label functions required extract multiple relationship types from the biomedical literature.
```

> To test the viability of re-use we implemented a sampling with replacement approach.

Reviewer: ```suggestion
We first established a baseline by training a generative model using only distant supervision label functions for the target edge type.
As an example, for the gene-interacts-gene edge type we used label functions that returned a `1` if the pair of genes was included in (LIST THE DATABASES HERE).
We then compared models that also included text and domain-heuristic label functions to these.
We sampled with replacement from these text- and domain-heuristic label functions separately within edge types, across edge types, and from a pool of all label functions.
We compared within-edge-type performance to across-edge-type and all-edge-type performance.
We performed each sampling for XYZ evenly-spaced numbers of label functions for each comparison from one to the total number of possible label functions.
We repeated the sampling process 50 times for each point.
We evaluated both generative and discriminative models at each point, and we recorded the performance of each in terms of the area under the receiver operating characteristic curve (AUROC) and the area under the precision-recall curve (AUPR).
```

Reviewer: I think you should make these changes and then go back to revise a bit more. I added AUPR since you had been presenting results for both. Did those go away?

Author: These did not go away. I incorporated them into the results section.

> For each relationship we trained a generative model using only label functions in the database category.

Reviewer 2: "database category" seems unclear? Do you mean edge type / metaedge? Let's get a consistent terminology.

These models surve as our baseline.
Following the baseline models, we randomly sample a fixed number of label functions (five evenly spaced points between one and the total number of constructed functions) from both the text patterns and domain heuristics categories.
These sampled functions are incorporated on top of the baseline models.

> We record performance of each sample in terms of area under the receiver operating curve (AUROC).

Reviewer 2: ```suggestion
We record performance of each sample in terms of area under the receiver operating characteristic curve (AUROC).
```
